velox/tool/trace/TableWriterReplayer.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <utility>
+
+#include "velox/tool/trace/OperatorReplayerBase.h"
+
+#include "velox/core/PlanNode.h"
+
+namespace facebook::velox::tool::trace {
+/// The tool used to print or replay the traced query metadata and operations.
+class TableWriterReplayer final : public OperatorReplayerBase {
+ public:
+  TableWriterReplayer(
+      const std::string& rootDir,
+      const std::string& taskId,
+      const std::string& nodeId,
+      const int32_t pipelineId,
+      const std::string& operatorType,
+      std::string targetDir)
+      : OperatorReplayerBase(rootDir, taskId, nodeId, pipelineId, operatorType),
+        targetDir_(std::move(targetDir)) {
+    VELOX_CHECK(!targetDir_.empty());
+  }
+
+  RowVectorPtr run() const override;
+
+  core::PlanNodePtr createPlan() const override;
+
+ private:
+  const std::string targetDir_;
```

Comment:
shall we name traceDir_ to be consistent with naming in other places?

Reply:
`targetDir_` is the output directory when we replay the `TableWriter`. Rename it to `tableWriterOutputDir_`?

Reply:
Renamed it to `replayOutputDir_`